### ISSUE TITLE
Order list of classes when scanning annotations

### ISSUE THREAD

**Issue by knutwannheden** [smallrye/smallrye-open-api#735]
I develop a Quarkus application where I use SmallRye OpenAPI to generate an OpenAPI definition. Now it is bugging me that every time I build the application the generated `openapi.yaml` file, the list of schemas can have a different order. This bugs me because I use the generated file in a Terraform resource for an Azure API Management API definition. Terraform picks up on the difference and consequently also updates the resource while applying the Terraform script.

I have tracked this down to `io.smallrye.openapi.jaxrs.JaxRsAnnotationScanner#processResourceClasses()` which calls `getJaxRsResourceClasses()` and the result is not sorted and can apparently have any order. So I am thinking if it would make sense to change `getJaxRsResourceClasses()` to apply a `sorted(Comparator<ClassInfo>)` operation which sorts the objects by name (fully qualified class name).

**Comment by MikeEdgar** [comment]
Hi @knutwannheden - just to clarify, you're referring to the entries in `/components/schemas` of the final definition, correct?

Note that this could also be accomplished using a filter. Rather than sorting by the origin of the schema (using `ClassInfo`), ordering based on the final name is probably closer to what you're looking for.

```java
class SortedSchemaFilter implements OASFilter {
    public void filterOpenAPI(OpenAPI openAPI) {
        openAPI.getComponents().setSchemas(new TreeMap<>(openAPI.getComponents.getSchemas()));
    }
}
```

**Comment by MikeEdgar** [comment]
The reason that sorting the entries in `components` has been avoided in the past is to allow the order given in a static `openapi.yaml` file or `OASModelReader` to be preserved.

**Comment by knutwannheden** [comment]
Hi @MikeEdgar - Yes, you are correct, this is about `/components/schemas`. I will give your suggested approach a go. We can then go ahead and close this issue again, if that works. Thanks for the pointer!

**Comment by knutwannheden** [comment]
Thinking a bit more about it, wouldn't it still be possible to implement the sorting as I proposed? It would only sort the input classes to be scanned. That should AFAICT not affect the order in which static content is added, as the in-memory document preserves the order in which elements are added to it. So I think the idea as I proposed it should not cause any problems with the static entries.
